doc/releases/changelog-0.23.0.md

```diff
@@ -140,59 +195,118 @@
   `method='pyscf'` in :func:`~.molecular_hamiltonian`.
 
   ```python
-  import pennylane as qml
-  from pennylane import numpy as np
-  
   symbols = ["H", "H"]
   geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 2.0]])
   hamiltonian, qubits = qml.qchem.molecular_hamiltonian(symbols, geometry, method='pyscf')
   ```
 
+  Summary of QChem changes:
```

Comment:
Some of these sounds like they should be proper, elaborated, changelog entries. In particular:

- New functions are added for computing multipole moment molecular integrals
- New functions are added for building a differentiable dipole moment observable
- New functions are added for building fermionic and qubit observables
- A new module is created for hosting openfermion to pennylane observable conversion functions
- The tapering functions are added to qchem
- The dependency on openbabel is removed
- The efficiency of computing molecular integrals and Hamiltonian is improved
- The 6-31G basis set is added to the qchem basis set repo

@soranjh I'm not sure the best place for them, but maybe either New Features or Improvements as you see appropriate?

They sound like _major_ changes that readers would want to know about!

I would remove the following, since these do not sound like important information for the release notes:

- Expressive names are used for the Hartree-Fock solver functions [This sounds like an implementation detail change?]
- Differentiable and non-differentiable backends can be selected for building a Hamiltonian [already mentioned]
- The quantum chemistry functionalities are unified [already mentioned]
- The qchem and new hf modules are merged [already mentioned]
- These new additions are added to a feature branch [not a change/feature 🙂]